Source/OxyPlot/Series/ExtrapolationLineSeries.cs

```diff
@@ -0,0 +1,322 @@
+    /// the intervals the style is always solid.
+    /// </summary>
+    public class ExtrapolationLineSeries : LineSeries
+    {
+        /// <summary>
+        /// Default color for the extrapolated parts of the curve. Currently hard-coded.
+        /// </summary>
+        private readonly OxyColor defaultExtrapolationColor = OxyColors.Black;
+
+        /// <summary>
+        /// Default line style for the extrapolated parts of the curve. Currently hard-coded.
+        /// </summary>
+        private readonly LineStyle defaultExtrapolationLineStyle = LineStyle.Dash;
+
+        private List<DataRange> orderedIntervals;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref = "ExtrapolationLineSeries" /> class.
+        /// </summary>
+        public ExtrapolationLineSeries()
+        {
+            this.ExtrapolationColor = OxyColors.Black;
+            this.LineStyle = LineStyle.Dot;
+        }
+
+        /// <summary>
+        /// Gets or sets the color for the part of the line that is inside an interval.
+        /// </summary>
+        public OxyColor ExtrapolationColor { get; set; }
+
+        /// <summary>
+        /// Gets the actual extrapolation color.
+        /// </summary>
+        /// <value>The actual color.</value>
+        public OxyColor ActualExtrapolationColor => this.ExtrapolationColor.GetActualColor(this.defaultExtrapolationColor);
+
+        /// <summary>
+        /// Gets or sets the style for the extrapolated parts of the line.
+        /// </summary>
+        public LineStyle ExtrapolationLineStyle { get; set; }
+
+        /// <summary>
+        /// Gets the actual extrapolation line style.
+        /// </summary>
+        public LineStyle ActualExtrapolationLineStyle
+        {
+            get
+            {
+                return this.ExtrapolationLineStyle != LineStyle.Automatic ?
+                    this.ExtrapolationLineStyle :
+                    this.defaultExtrapolationLineStyle;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the extrapolated regions of the series will
+        /// be taken into account when calculating the minima and maxima of the dataset.
+        /// These regions will hence also be ignored when auto-scaling the axes.
+        /// </summary>
+        public bool IgnoreExtraplotationForScaling { get; set; }
+
+        /// <summary>
+        /// Gets the list of X intervals within which the line is rendered using the second color and style.
+        /// </summary>
+        public List<DataRange> Intervals { get; } = new List<DataRange>();
+
+        /// <summary>
+        /// Gets the actual dash array for the extrapolated parts of the line.
+        /// </summary>
+        protected double[] ActualExtrapolationDashArray
+        {
+            get
+            {
+                return this.Dashes ?? this.ActualExtrapolationLineStyle.GetDashArray();
```

Comment:
This means it will use the 'normal' dash style if it was set: we should probably provide another `ExtrapolationLineDashes`.

Reply:
Done. Also added a new example to test it to the browser.